pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -638,16 +910,138 @@ func (pl *dynamicResources) PreFilter(ctx context.Context, state *framework.Cycl
 				}
 				s.informationsForClaim[index].availableOnNode = selector
 			}
-			// Now we need information from drivers.
 			s.informationsForClaim[index].status = statusForClaim(s.podSchedulingState.schedulingCtx, pod.Spec.ResourceClaims[index].Name)
+
+			if class.StructuredParameters {
+				s.informationsForClaim[index].structuredParameters = true
+
+				// Allocation in flight? Better wait for that
+				// to finish, see inFlightAllocations
+				// documentation for details.
+				if _, found := pl.inFlightAllocations.Load(claim.UID); found {
+					return nil, statusUnschedulable(logger, fmt.Sprintf("resource claim %s is in the process of being allocated", klog.KObj(claim)))
+				}
+
+				// We need the claim and class parameters. If
+				// they don't exist yet, the pod has to wait.
+				//
+				// TODO (https://github.com/kubernetes/kubernetes/issues/123697):
+				// check this already in foreachPodResourceClaim, together with setting up informationsForClaim.
+				// Then PreEnqueue will also check for existence of parameters.
+				classParameters, claimParameters, status := pl.lookupParameters(logger, class, claim)
+				if status != nil {
+					return nil, status
+				}
+				controller, err := newClaimController(logger, class, classParameters, claimParameters)
+				if err != nil {
+					return nil, statusError(logger, err)
+				}
+				s.informationsForClaim[index].controller = controller
+				needResourceInformation = true
+			} else if claim.Spec.AllocationMode == resourcev1alpha2.AllocationModeImmediate {
+				// This will get resolved by the resource driver.
+				return nil, statusUnschedulable(logger, "unallocated immediate resourceclaim", "pod", klog.KObj(pod), "resourceclaim", klog.KObj(claim))
+			}
+		}
+	}
+
+	if needResourceInformation {
```

Comment:
It might get set multiple times if there are several claims which need resource information. We only want to do `newResourceModel` once, therefore the boolean and this if check are needed.